Mathlib/RingTheory/MvPolynomial/Homogeneous.lean

```diff
@@ -34,19 +35,34 @@ variable {σ : Type*} {τ : Type*} {R : Type*} {S : Type*}
 
 /-
 TODO
-* create definition for `∑ i in d.support, d i`
 * show that `MvPolynomial σ R ≃ₐ[R] ⨁ i, homogeneousSubmodule σ R i`
 -/
+
+/-- The degree of a monomial. -/
+def degree (d : σ →₀ ℕ) := ∑ i in d.support, d i
+
+theorem degree_eq_weightedDegree (d : σ →₀ ℕ) :
+    ∑ i in d.support, d i = weightedDegree 1 d := by
+  simp [weightedDegree, Finsupp.total, Finsupp.sum]
+
 /-- A multivariate polynomial `φ` is homogeneous of degree `n`
 if all monomials occurring in `φ` have degree `n`. -/
 def IsHomogeneous [CommSemiring R] (φ : MvPolynomial σ R) (n : ℕ) :=
-  ∀ ⦃d⦄, coeff d φ ≠ 0 → ∑ i in d.support, d i = n
+  IsWeightedHomogeneous 1 φ n
 #align mv_polynomial.is_homogeneous MvPolynomial.IsHomogeneous
 
+variable [CommSemiring R]
+
+theorem totalDegree_eq_weightedTotalDegree (φ : MvPolynomial σ R) :
+    φ.totalDegree = weightedTotalDegree (1 : σ → ℕ) φ := by
+  simp only [totalDegree, weightedTotalDegree, weightedDegree, LinearMap.toAddMonoidHom_coe,
+    Finsupp.total, Pi.one_apply, Finsupp.coe_lsum, LinearMap.coe_smulRight, LinearMap.id_coe,
+    id.def, Algebra.id.smul_eq_mul, mul_one]
+
 variable (σ R)
 
 /-- The submodule of homogeneous `MvPolynomial`s of degree `n`. -/
-def homogeneousSubmodule [CommSemiring R] (n : ℕ) : Submodule R (MvPolynomial σ R) where
+def homogeneousSubmodule (n : ℕ) : Submodule R (MvPolynomial σ R) where
```

Comment:
For completeness, could you also add a (`@[simp]`?) lemma stating `weightedHomogeneousSubmodule 1 R n = homogeneousSubmodule σ R n`?